app/generic/verifier/verifier.py

```diff
@@ -36,6 +39,103 @@ def __get_verifier_by_version(protocol_version: str) -> Verifier:
         raise ValueError(f"Unknown protocol version {protocol_version}")
 
 
+@router.post("/credentials")
+async def get_credentials(
+    proof_request: ProofRequestGeneric,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> List[IndyCredPrecis]:
+    """
+     Get matching credentials for  presentation exchange
+
+     Parameters:
+     ----------
+    proof_request: ProofRequestGeneric
+         The proof request object
+
+     Returns:
+     --------
+     presentation_exchange_list: [IndyCredPrecis]
+         The list of Indy presentation credentials
+    """
+    try:
+        prover = __get_verifier_by_version(
+            protocol_version=proof_request.protocol_version
+        )
+        return await prover.get_creds(
+            controller=aries_controller, pres_ex_id=proof_request.proof_id
+        )
+    except Exception as e:
+        logger.error(
+            f"Failed to get matching credentials: \n {proof_request.proof_id}\n \n{e!r}"
+        )
+        raise e from e
+
+
+@router.post("/proofs")
+async def get_proofs(
```

Comment:
Same as with the acapy_verifier I would split this up in multiple endpoints (get proof by id and get proofs).